### PR TITLE
feat: konflux: test ocp upgrade periodically

### PR DESCRIPTION
**Pull request by psturc**
# Description
[KONFLUX-2063](https://issues.redhat.com//browse/KONFLUX-2063)

* added a periodic (nightly) job that performs the following steps:
1. Claims a cluster
2. Installs Konflux
3. Performs upgrade to a next minor version of openshift (e.g. 4.13 -> 4.14)
4. Runs rhtap(konflux)-demo test suite

# [Link to a successful run](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/48693/rehearse-48693-periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic/1758458619818414080)

### CODE REVIEW

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic
/test all

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-e2e-tests-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

**Comment by psturc** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-konflux-openshift-upgrade-periodic

File: ci-operator/step-registry/redhat-appstudio/openshift-upgrade/redhat-appstudio-openshift-upgrade-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # https://docs.ci.openshift.org/docs/architecture/step-registry/#marking-post-steps-best-effort
+    allow_best_effort_post_steps: true
+    # https://docs.ci.openshift.org/docs/architecture/step-registry/#skipping-post-steps-on-success
+    allow_skip_on_success: true
+    env:
+      # for "openshift-configure-cincinnati" step
+      CHANGE_CHANNEL_BASE: candidate
+      # for "redhat-appstudio-e2e" step
+      SKIP_BOOTSTRAP: "true"
+      E2E_TEST_SUITE_LABEL: "rhtap-demo"
```
Comment (tisutisu) [inline ci-operator/step-registry/redhat-appstudio/openshift-upgrade/redhat-appstudio-openshift-upgrade-workflow.yaml:13]: Thanks for the upgrade test automation. Running `rhtap-demo` suite after the upgrade complete will be testing there is no issue creating user workloads after the upgrade. Will it be better to also tests there is no impact for the existing users ? I mean creating user workloads before upgrade and see if those user workloads are not impacted by the upgrade.

Comment (psturc) [reply ci-operator/step-registry/redhat-appstudio/openshift-upgrade/redhat-appstudio-openshift-upgrade-workflow.yaml:13]: good idea, I created a [follow-up ticket for this](https://issues.redhat.com/browse/KONFLUX-2269)

**Pull request merged**